NEWS.md

```diff
@@ -3,6 +3,306 @@ XGBoost Change Log
+
+### Parameter validation: detection of unused or incorrect parameters (#4553, #4577, #4738, #4801, #4961, #5101, #5157, #5167, #5256)
+* Mis-spelled training parameter is a common user mistake. In previous versions of XGBoost, mis-spelled parameters were silently ignored. Starting with 1.0.0 release, XGBoost will produce a warning message if there is any unused training parameters.
+* Configuration steps now have well-defined semantics (#4542, #4738), so we know exactly where and how the internal configurable parameters are changed.
+* The user can now use `get_params()` function to inspect all training parameters. This is helpful for debugging model performance.
+
+### Allow individual workers to recover from faults (#4808, #4966)
+* Status quo: if a worker fails, all workers are shut down and restarted, and learning resumes from the last checkpoint. This involves requesting resources from the scheduler (e.g. Spark) and shuffling all the data again from scratch. Both of these operations can be quite costly and block training for extended periods of time, especially if the training data is big and the number of worker nodes is in the hundreds.
+* The proposed solution is to recover the single node that failed, instead of shutting down all workers. The rest of the clusters wait until the single failed worker is bootstrapped and catches up with the rest.
+* See roadmap at #4753. Note that this is work in progress. In particular, the feature is not yet available from XGBoost4J-Spark.
+
+### Accurate prediction for DART models
+* Use DART tree weights when computing SHAPs (#5050)
+* Don't drop trees during DART prediction by default (#5115)
+* Fix DART prediction in R (#5204)
+
+### Make external memory more robust
+* Fix issues with training with external memory on cpu (#4487)
+* Fix crash with approx tree method on cpu (#4510)
+* Fix external memory race in `exact` (#4980). Note: `dmlc::ThreadedIter` is not actually thread-safe. We would like to re-design it in the long term.
+
+### Major refactoring of the `DMatrix` class (#4686, #4744, #4748, #5044, #5092, #5108, #5188, #5198)
+* Goal 1: improve performance and reduce memory consumption. Right now, if the user trains a model with a NumPy array as training data, the array gets copies 2-3 times before training begins. We'd like to reduce duplication of the data matrix.
+* Goal 2: Expose a common interface to external data, unify the way DMatrix objects are constructed and simplify the process of adding new external data sources. This work is essential for ingesting cuPy arrays.
+* RFC: #4354, Roadmap: #5143
+* This work is also relevant to external memory support on GPUs.
+
+### Breaking: XGBoost Python package now requires Python 3.5 or newer (#5021, #5274)
```

Comment:
This is the version we are sunseting the make files.  Might be worth nothing as a breaking change.